include/swift/AST/ASTBridging.h

```diff
@@ -82,14 +175,761 @@ class BridgedDiagnosticFixIt {
+           "genericWhereClause:rightAngleLoc:)")
+BridgedGenericParamList BridgedGenericParamList_createParsed(
+    BridgedASTContext cContext, BridgedSourceLoc cLeftAngleLoc,
+    BridgedArrayRef cParameters,
+    BridgedNullableTrailingWhereClause genericWhereClause,
+    BridgedSourceLoc cRightAngleLoc);
+
+SWIFT_NAME(
+    "BridgedGenericTypeParamDecl.createParsed(_:declContext:eachKeywordLoc:"
+    "name:nameLoc:inheritedType:index:)")
+BridgedGenericTypeParamDecl BridgedGenericTypeParamDecl_createParsed(
+    BridgedASTContext cContext, BridgedDeclContext cDeclContext,
+    BridgedSourceLoc cEachLoc, BridgedIdentifier cName,
+    BridgedSourceLoc cNameLoc, BridgedNullableTypeRepr opaqueInheritedType,
+    size_t index);
+
+SWIFT_NAME(
+    "BridgedTrailingWhereClause.createParsed(_:whereKeywordLoc:requirements:)")
+BridgedTrailingWhereClause
+BridgedTrailingWhereClause_createParsed(BridgedASTContext cContext,
+                                        BridgedSourceLoc cWhereKeywordLoc,
+                                        BridgedArrayRef cRequirements);
+
+SWIFT_NAME("BridgedParameterList.createParsed(_:leftParenLoc:parameters:"
+           "rightParenLoc:)")
+BridgedParameterList BridgedParameterList_createParsed(
+    BridgedASTContext cContext, BridgedSourceLoc cLeftParenLoc,
+    BridgedArrayRef cParameters, BridgedSourceLoc cRightParenLoc);
+
+//===----------------------------------------------------------------------===//
+// MARK: Plugins
+//===----------------------------------------------------------------------===//
+
+SWIFT_BEGIN_ASSUME_NONNULL
+
+typedef void *PluginHandle;
+typedef const void *PluginCapabilityPtr;
+
+/// Set a capability data to the plugin object. Since the data is just a opaque
+/// pointer, it's not used in AST at all.
+void Plugin_setCapability(PluginHandle handle,
+                          PluginCapabilityPtr _Nullable data);
+
+/// Get a capability data set by \c Plugin_setCapability .
+PluginCapabilityPtr _Nullable Plugin_getCapability(PluginHandle handle);
+
+/// Get the executable file path of the plugin.
+const char *Plugin_getExecutableFilePath(PluginHandle handle);
+
+/// Lock the plugin. Clients should lock it during sending and recving the
+/// response.
+void Plugin_lock(PluginHandle handle);
+
+/// Unlock the plugin.
+void Plugin_unlock(PluginHandle handle);
+
+/// Launch the plugin if it's not running.
+bool Plugin_spawnIfNeeded(PluginHandle handle);
+
+/// Sends the message to the plugin, returns true if there was an error.
+/// Clients should receive the response  by \c Plugin_waitForNextMessage .
+bool Plugin_sendMessage(PluginHandle handle, const BridgedData data);
+
+/// Receive a message from the plugin.
+bool Plugin_waitForNextMessage(PluginHandle handle, BridgedData *data);
+
+SWIFT_END_ASSUME_NONNULL
 
 SWIFT_END_NULLABILITY_ANNOTATIONS
 
+#ifndef PURE_BRIDGING_MODE
+// In _not_ PURE_BRIDGING_MODE, briding functions are inlined and therefore
```

Comment:
Because the bridging functions can require C++ headers that we don't want to include in "pure bridging mode", since that can trigger the Windows and LLDB C++ interop issues. I can update the comment to explain this (FWIW I copied it from SILBridging.h)